sumpy/p2e.py

```diff
@@ -44,30 +45,30 @@
 
 # {{{ P2E base class
 
-class P2EBase(KernelCacheWrapper):
-    def __init__(self, ctx, expansion,
-            options=[], name=None, device=None):
+class P2EBase(KernelComputation, KernelCacheWrapper):
+    """Common input processing for kernel computations."""
+
+    def __init__(self, ctx, expansion, kernels=None,
+            options=[], name=None, device=None, strength_usage=None):
         """
         :arg expansion: a subclass of :class:`sympy.expansion.ExpansionBase`
         :arg strength_usage: A list of integers indicating which expression
           uses which source strength indicator. This implicitly specifies the
           number of strength arrays that need to be passed.
           Default: all kernels use the same strength.
         """
+        if kernels is None:
+            kernels = [expansion.kernel]
```

Comment:
```suggestion
            kernels = [expansion.kernel]
        if options is None:
            options = []
```
(maybe this wants to be in `KernelComputation` instead.)

Reply:
`options` is not used anywhere, right? Shall I remove it?

Reply:
I'm OK with that.